src/System.Diagnostics.Process/tests/ProcessTests.OSX.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Security;
+using Xunit;
+using Xunit.NetCore.Extensions;
+
+namespace System.Diagnostics.Tests
+{
+    public partial class ProcessTests : ProcessTestBase
+    {
+        [Fact]
+        [OuterLoop("Launches default application")]
+        public void TestWithFilename_ShouldUseOpenWithDefaultApp()
+        {
+            string file = Path.Combine(Environment.CurrentDirectory, "..", "..", "..", "..", "PATENTS.TXT");
+            using (var px = Process.Start("/usr/bin/open", file))
+            {
+                Assert.False(px.HasExited);
+                px.WaitForExit();
+                Assert.True(px.HasExited);
+                Assert.Equal(0, px.ExitCode); // Exit Code 0 from open means success
+            }
+        }
+
+        [Fact]
+        [OuterLoop("Launches default browser")]
+        public void TestWithUrl_ShouldUseOpenWithDefaultApp()
+        {
+            using (var px = Process.Start("/usr/bin/open", "http://www.google.com"))
```

Comment:
why is /usr/bin/open relevant here? does xdg-open etc not exist on OSX? and if it does not, should we be looking for /usr/bin/open?
On Windows, `Process.Start("http://bing.com")` pops the browser. Ideally that would work on OSX and Linux just the same. Doesn't it in Mono?

Reply:
I think `Process.Start("http://bing.com")` won't work, as `UseShellExecute` is AFAIK false by default in dotnet core. (Maybe the default could be changed to true after `UseShellExecute` is correctly implemented on all platforms?(via this pull request)) I agree, that `/usr/bin/open` should not be used explicitly, instead `UseShellExecute` should be used.

Reply:
Youre right, I didn't mean literally, I meant without usr bin open.